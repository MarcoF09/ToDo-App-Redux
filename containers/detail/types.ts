import {
  NavigationParams,
  NavigationRoute,
  NavigationScreenProp
} from 'react-navigation'
import { Todo } from '../../types/globalTypes'

export interface Props {
  todo: Todo[]
  markAsNotDone: (todos: Todo[], index: number) => void
  markAsDone: (todos: Todo[], index: number) => void
  navigation: NavigationScreenProp<
    NavigationRoute<NavigationParams>,
    NavigationParams
  >
}

export type LayoutProps = Props
