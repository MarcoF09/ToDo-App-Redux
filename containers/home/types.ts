import { Animated } from 'react-native';
import { NavigationRoute, NavigationScreenProp } from 'react-navigation';
import { State, Todo } from '../../types/globalTypes';

export interface Props {
  todo: Todo[];
  navigation: NavigationScreenProp<NavigationRoute>;
  getToDoData: () => void;
  clearAllDone: (todos: Todo[]) => void;
  changeCheckBoxState: (todos: Todo[], index: number) => void;
  deleteItem: (todos: Todo[], index: number) => void;
  animateDeleteButton: (
    deleteButtonState: boolean,
    moveAnimation: Animated.ValueXY
  ) => void;
}

export interface HomeState extends State {
  deleteButtonState: boolean;
  moveAnimation: Animated.ValueXY;
}

export type LayoutProps = Props & HomeState;
