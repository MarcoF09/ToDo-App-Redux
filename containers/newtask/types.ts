import { NavigationRoute, NavigationScreenProp } from 'react-navigation';

interface Props {
  navigation: NavigationScreenProp<NavigationRoute>;
  addToDo: (title: string, description: string) => void;
  onChangeTitle: (text: string) => void;
  onChangeDescription: (text: string) => void;
  onFocusTitle: () => void;
  onBlurTitle: () => void;
  onFocusDescription: () => void;
  onBlurDescription: () => void;
}

interface NewTaskState {
  firstInputColor: string;
  secondInputColor: string;
  title: string;
  description: string;
}

export type LayoutProps = Props & NewTaskState;
