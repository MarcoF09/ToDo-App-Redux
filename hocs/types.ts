import React from 'react';

export type Hoc<TOuter = {}, TInner = TOuter> = (
  Component: React.ComponentType<TInner>
) => React.ComponentType<TOuter>;
export type ExtractTInner<T> = T extends Hoc<any, infer TInner>
  ? TInner
  : never;
