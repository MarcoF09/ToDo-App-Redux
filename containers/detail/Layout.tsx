import React, { FunctionComponent } from 'react';
import { StatusBar, Text, View } from 'react-native';
import { Colors } from '../../colors/Colors';
import { Button } from '../../components/button';
import { Strings } from '../../strings';
import { styles } from './styles';
import { LayoutProps } from './types';

export const Layout: FunctionComponent<LayoutProps> = ({
  markAsDone,
  todo,
  navigation,
  markAsNotDone
}: LayoutProps) => {
  return (
    <View style={styles.container}>
      <StatusBar backgroundColor={Colors.customBlue} />
      <View style={styles.topButtonContainer}>
        <Button
          text={Strings.notDone}
          onPress={() => {
            markAsNotDone(todo, navigation.getParam('index'));
            navigation.pop();
          }}
        />
      </View>
      <Text style={styles.primaryText}>
        {todo[navigation.getParam('index')].title}
      </Text>
      <Text style={styles.secondaryText}>
        {todo[navigation.getParam('index')].description}
      </Text>
      <Button
        text={Strings.markAsDone}
        onPress={() => {
          markAsDone(todo, navigation.getParam('index'));
          navigation.pop();
        }}
      />
    </View>
  );
};
