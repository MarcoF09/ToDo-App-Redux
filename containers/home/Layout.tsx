import React, { FunctionComponent } from 'react'
import { FlatList, StatusBar, View } from 'react-native'
import { Colors } from '../../colors/Colors'
import { renderFooter } from './renderFooter'
import { renderItem } from './renderItem'
import { LayoutProps } from './types'

export const Layout: FunctionComponent<LayoutProps> = ({
  todo,
  navigation,
  clearAllDone,
  moveAnimation,
  deleteItem,
  changeCheckBoxState
}: LayoutProps) => {
  return (
    <View testID="homeLayout">
      <StatusBar backgroundColor={Colors.customBlue} />
      <FlatList
        data={todo}
        keyExtractor={(item, index) => index.toString()}
        renderItem={item =>
          renderItem(
            item,
            navigation,
            todo,
            moveAnimation,
            deleteItem,
            changeCheckBoxState
          )
        }
        ListFooterComponent={() => renderFooter(todo, clearAllDone)}
      />
    </View>
  )
}
