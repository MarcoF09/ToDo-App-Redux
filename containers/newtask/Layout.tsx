import React, { FunctionComponent } from 'react'
import { StatusBar, TextInput, View } from 'react-native'
import { Colors } from '../../colors/Colors'
import { Strings } from '../../strings'
import { styles } from './styles'
import { LayoutProps } from './types'

export const Layout: FunctionComponent<LayoutProps> = ({
  firstInputColor,
  secondInputColor,
  onFocusTitle,
  onBlurTitle,
  onFocusDescription,
  onBlurDescription,
  onChangeTitle,
  onChangeDescription
}: LayoutProps) => {
  return (
    <View style={styles.container} testID="layout">
      <StatusBar backgroundColor={Colors.customBlue} />
      <TextInput
        testID="title"
        onFocus={() => onFocusTitle()}
        onBlur={() => onBlurTitle()}
        style={{
          borderBottomWidth: 1,
          borderBottomColor: firstInputColor,
          fontSize: 36,
          color: Colors.black,
          fontFamily: 'SourceSansPro-Regular'
        }}
        placeholder={Strings.titlePlaceHolder}
        multiline={false}
        onChangeText={text => onChangeTitle(text)}
      />
      <TextInput
        testID="description"
        onFocus={() => onFocusDescription()}
        onBlur={() => onBlurDescription()}
        style={{
          borderBottomWidth: 1,
          borderBottomColor: secondInputColor,
          textAlignVertical: 'top',
          height: 90,
          fontFamily: 'SourceSansPro-Regular',
          fontSize: 14
        }}
        placeholder={Strings.descriptionPlaceHolder}
        multiline={true}
        numberOfLines={5}
        onChangeText={text => onChangeDescription(text)}
      />
    </View>
  )
}
