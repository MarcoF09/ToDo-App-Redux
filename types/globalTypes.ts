export interface Todo {
  id?: string;
  title: string;
  order?: number;
  description?: string;
  completed: boolean;
  url: string;
}

export interface ResponseBackend {
  title: string;
  order: number;
  completed: boolean;
  url: string;
}

export interface State {
  todo: Todo[];
}
