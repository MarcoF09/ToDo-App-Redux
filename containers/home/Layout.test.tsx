import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import React from 'react'
import { Animated } from 'react-native'
import { Todo } from 'types/globalTypes'
import { Layout } from './Layout'

const createTestProps = (props: object) => ({
  navigation: {
    navigate: jest.fn()
  },
  ...props
})

describe('should render home screen', () => {
  const mockedFunction = jest.fn()
  const initialTodo: Todo[] = [
    { title: 'Title1', description: 'Description1', completed: false, url: '' },
    { title: 'Title2', description: 'Description2', completed: true, url: '' }
  ]
  const moveAnimation = new Animated.ValueXY({ x: -100, y: 30 })
  let props: any

  it('renders correctly without delete button', () => {
    props = createTestProps({ mockedFunction, initialTodo, moveAnimation })
    const component = shallow(
      <Layout
        todo={props.initialTodo}
        navigation={props.navigation}
        clearAllDone={props.mockedFunction}
        moveAnimation={props.moveAnimation}
        deleteItem={props.mockedFunction}
        changeCheckBoxState={props.mockedFunction}
        getToDoData={props.mockedFunction}
        animateDeleteButton={props.mockedFunction}
        deleteButtonState={false}
      />
    )
    expect(toJson(component)).toMatchSnapshot()
  })

  it('renders correctly with delete button', () => {
    props = createTestProps({ mockedFunction, initialTodo, moveAnimation })
    const component = shallow(
      <Layout
        todo={props.initialTodo}
        navigation={props.navigation}
        clearAllDone={props.mockedFunction}
        moveAnimation={props.moveAnimation}
        deleteItem={props.mockedFunction}
        changeCheckBoxState={props.mockedFunction}
        getToDoData={props.mockedFunction}
        animateDeleteButton={props.mockedFunction}
        deleteButtonState={true}
      />
    )
    expect(toJson(component)).toMatchSnapshot()
  })

  it('renders correctly without todos', () => {
    props = createTestProps({ mockedFunction, moveAnimation })
    const component = shallow(
      <Layout
        todo={props.initialTodo}
        navigation={props.navigation}
        clearAllDone={props.mockedFunction}
        moveAnimation={props.moveAnimation}
        deleteItem={props.mockedFunction}
        changeCheckBoxState={props.mockedFunction}
        getToDoData={props.mockedFunction}
        animateDeleteButton={props.mockedFunction}
        deleteButtonState={true}
      />
    )
    expect(toJson(component)).toMatchSnapshot()
  })
})
