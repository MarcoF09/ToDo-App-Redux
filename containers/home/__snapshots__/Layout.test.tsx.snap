// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render home screen renders correctly with delete button 1`] = `
<View>
  <StatusBar
    animated={false}
    backgroundColor="#1f86ff"
    showHideTransition="fade"
  />
  <FlatList
    ListFooterComponent={[Function]}
    data={
      Array [
        Object {
          "completed": false,
          "description": "Description1",
          "title": "Title1",
          "url": "",
        },
        Object {
          "completed": true,
          "description": "Description2",
          "title": "Title2",
          "url": "",
        },
      ]
    }
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
</View>
`;

exports[`should render home screen renders correctly without delete button 1`] = `
<View>
  <StatusBar
    animated={false}
    backgroundColor="#1f86ff"
    showHideTransition="fade"
  />
  <FlatList
    ListFooterComponent={[Function]}
    data={
      Array [
        Object {
          "completed": false,
          "description": "Description1",
          "title": "Title1",
          "url": "",
        },
        Object {
          "completed": true,
          "description": "Description2",
          "title": "Title2",
          "url": "",
        },
      ]
    }
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
</View>
`;

exports[`should render home screen renders correctly without todos 1`] = `
<View>
  <StatusBar
    animated={false}
    backgroundColor="#1f86ff"
    showHideTransition="fade"
  />
  <FlatList
    ListFooterComponent={[Function]}
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
</View>
`;
